client/components/Product_Overview/AddToCart.jsx

```diff
@@ -58,48 +58,121 @@ const ButtonStyle = styled.button`
+    if (storedOutfits) {
+      setOutfitsArray(storedOutfits);
+      setOutfitsStylesObj(storedStyles);
+    }
+  };
 
   useEffect(() => {
-    getResults();
-  }, [productOverviewId]);
+    getYourOutfits();
+  }, []);
 
-  const handleChange = ((event) => {
-    setAmount(event.target.value);
-  });
-  console.log(results)
+  const onAddCardClickHandler = () => {
+    axios.get('/product-features', { params: { id: productOverviewId } })
+      .then(({ data }) => {
+        const newOutfitsArray = outfitsArray;
+        const newOutfitsStylesObj = outfitsStylesObj || {};
+        let duplicate = false;
+
+        setCurrentProductData(data);
+
+        newOutfitsArray.forEach((outfit) => {
+          if (outfit.id === data.id) {
+            duplicate = true;
+          }
+        });
+
+        if (!duplicate) {
+          newOutfitsArray.push(data);
+          setOutfitsArray(newOutfitsArray);
+          setCurrentProductData(data);
+        } else {
+          alert('This item is already saved in your outfits!');
+        }
+
+        axios.get('/outfit-styles', { params: { id: productOverviewId } })
+          .then(({ data }) => {
+            newOutfitsStylesObj[data.product_id] = data.results;
+            setOutfitsStylesObj(newOutfitsStylesObj);
+
+            return {
+              outfitInfo: newOutfitsArray,
+              outfitStyles: newOutfitsStylesObj,
+            };
+          })
+          .then(({ outfitInfo, outfitStyles }) => {
+            localStorage.setItem('outfits', JSON.stringify(outfitInfo));
+            localStorage.setItem('styles', JSON.stringify(outfitStyles));
+            getYourOutfits();
+          })
+          .catch((err) => console.log(err));
+      })
+      .catch((err) => console.log(err));
+  };
 
   return (
     <CartGrid>
       <Column1Row1>
-        <SelectStyle onChange={handleChange}>
+        <SelectStyle onChange={handleSizeChange}>
           <option>Select Size</option>
           {Object.keys(results).map((size) => (
-            <option key={size} value={results[size].quantity}>{results[size].size}</option>
+            results[size].quantity !== 0
+            && (
+              <option
+                key={size}
+                value={[size, results[size].quantity]}
+              >
+                {results[size].size}
+              </option>
+            )
           ))}
         </SelectStyle>
       </Column1Row1>
       <Column2Row1>
-        <SelectStyle>
+        <SelectStyle onChange={handleQuantityChange}>
           <option>-</option>
-          <option value="quantity">{amount}</option>
+          {quantityVals.map((quantity) => (
+            <option value={quantity}>{quantity}</option>
+          ))}
         </SelectStyle>
       </Column2Row1>
       <Column1Row2>
-        <ButtonStyle type="submit">ADD TO BAG</ButtonStyle>
+        <ButtonStyle type="submit" onClick={onAddCardClickHandler}>ADD TO BAG</ButtonStyle>
```

Comment:
@monkeymedic26 you sure you want to add the outfit click handler to your "add to bag" button...?

Reply:
no I don't. will fix now.
thanks lol